File: 969-number-of-recent-calls/number-of-recent-calls.cpp
class RecentCounter {
public:
    vector<int> buf;
    
    int ping(int t) {
        int c = 0;
        buf.push_back(t);
        for (auto n : buf)
            if (t - 3000 <= n && n <= t)
                c++;
        return c;
    }
};

/**
 * Your RecentCounter object will be instantiated and called as such:
 * RecentCounter* obj = new RecentCounter();
 * int param_1 = obj->ping(t);
 */

File: 969-number-of-recent-calls/number-of-recent-calls.go
type RecentCounter struct {
    calls []int
}


func Constructor() RecentCounter {
    return RecentCounter{}
}


func (this *RecentCounter) Ping(t int) int {
    this.calls = append(this.calls, t)
    countCalls := 0

    for _, v := range this.calls {
        if t - 3000 <= v && v <= t {
            countCalls++
        }
    }

    return countCalls
}


/**
 * Your RecentCounter object will be instantiated and called as such:
 * obj := Constructor();
 * param_1 := obj.Ping(t);
 */